src/sagemaker/base_deserializers.py

```diff
@@ -0,0 +1,324 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Implements base methods for deserializing data returned from an inference endpoint."""
+from __future__ import absolute_import
+
+import csv
+
+import abc
+import codecs
+import io
+import json
+
+import numpy as np
+from six import with_metaclass
+
+from sagemaker.utils import DeferredError
+
+try:
+    import pandas
+except ImportError as e:
+    pandas = DeferredError(e)
```

Comment:
Could we add a comment explaining the necessity of this deferred error please?

Reply:
this is all code that was moved from `deserializers.py`